dcc-portal-ui/app/scripts/survivalanalysis/js/service.js

```diff
@@ -0,0 +1,199 @@
+        return tsv;
+      }
+
+      _.extend(this, {
+        fetchSurvivalData: fetchSurvivalData,
+        dataSetToTSV: dataSetToTSV
+      });
+
+    });
+
+  module.service('SurvivalAnalysisLaunchService', function(SetService, Restangular, LocationService,
+    SetNameService, $timeout, $location, Page){
+      const _service = this;
+
+      var d1, d2, type = 'donor';
+
+      _service.getSetName = function(filters){
+        return SetNameService.getSetFilters()
+          .then(function (filters) {
+            return SetNameService.getSetName(filters);
+          });
+      }
+
+      // Wait for sets to materialize
+      _service.wait = (ids, numTries, callback) => {
+        if (numTries <= 0) {
+          return;
+        }
+        SetService.getMetaData(ids).then(function(data) {
+          var finished = _.filter(data, function(d) {
+            return d.state === 'FINISHED';
+          });
+
+          if (finished.length === ids.length) {
+            callback(data);
+          } else {
+            $timeout(function() {
+              _service.wait(ids, --numTries, callback);
+            }, 1500);
+          }
+        });
+      }
+
+       /* Phenotype comparison only takes in donor set ids */
+      _service.launchAnalysis = (setIds) => {
+        return Restangular
+          .one('analysis')
+          .post('phenotype', setIds, {}, {'Content-Type': 'application/json'})
+          .then(function(data) {
+            if (!data.id) {
+              console.log('Could not retrieve analysis data.id');
+            }
+            LocationService.goToPath('analysis/view/phenotype/' + data.id);
+          }).finally(function(){
+            Page.stopWork();
+          });
+      };
+
+      _service.launchSurvivalAnalysis = (entityType, entityId, entitySymbol, filters) => {
+
+        Page.startWork();
+
+        _service.getSetName(filters).then(function (setName) {
+          d1 = {
+            filters: entityType === 'gene' ? _.extend(_.clone(filters), {gene: { id: { is: [entityId] } }}) : _.extend(_.clone(filters), {mutation: { id: { is: [entityId] } }}),
+            isTransient: true,
+            type: type,
+            name: entityType === 'gene' ? entitySymbol + ' Mutated Donors ' + setName : 'Donors with ' + entitySymbol + ' ' + setName
+          };
+
+          d2 = {
+            filters: entityType === 'gene' ? _.extend(_.clone(filters), {gene: { id: { not: [entityId] } }}) : _.extend(_.clone(filters), {mutation: { id: { not: [entityId] } }}),
+            isTransient: true,
+            type: type,
+            name: entityType === 'gene' ? entitySymbol + ' Not Mutated Donors ' + setName : 'Donors without ' + entitySymbol + ' ' + setName
+          };
+
+          var setIds = [];
+
+          SetService.addSet(type, d1).then(function (r1) {
+            setIds.push(r1.id);
+            SetService.addSet(type, d2).then(function (r2) {
+              setIds.push(r2.id);
+              function proxyLaunch() {
```

Comment:
Don't see the need for such methods if they only contain one statement. Probably better to use an arrow function.